test/Libraries/RevitNodesTests/Elements/ElementTests.cs

```diff
@@ -436,5 +436,42 @@ private static void AssertElementsAreJoined(Element element, Element otherElemen
         }
 
         #endregion
+
+        [Test]
+        [TestModel(@".\Element\elementIntersection.rvt")]
+        public void CanSuccessfullyGetIntersectingElementsOfSpeceficCategory()
+        {
+            // Element to check intersections on
+            int intersectionElementId = 316167;
+            var intersectionElement = ElementSelector.ByElementId(intersectionElementId, true);
+
+            // Element intersecting
+            int structuralFramingId = 316318;
+            var structuralFramingElement = ElementSelector.ByElementId(structuralFramingId, true);
+            int floorId = 316539;
+            var floorElement = ElementSelector.ByElementId(floorId, true);
+            int wallId = 316246;
+            var wallElement = ElementSelector.ByElementId(wallId, true);
+
+            // Get intersecting elements of category
+            Revit.Elements.Category structuralFrameCategory = Revit.Elements.Category.ByName("StructuralFraming");
+            int intersectedFraming = intersectionElement.GetIntersectingElementsOfCategory(structuralFrameCategory).FirstOrDefault<Element>().Id;
```

Comment:
we should be getting all the elements intersecting and only at the end asserting there's only 1. there's potential for this to return unexpected ones and we need to allow test to check for that.

Reply:
this applies to all 3 categories